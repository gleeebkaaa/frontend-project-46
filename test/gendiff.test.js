import generateDiff from '../src/index.js';
import { fileURLToPath } from 'url';
import path from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const getFixturePath = (filename) => path.join(__dirname, '..', '__fixtures__', filename);

describe('generateDiff', () => {
    test('добавление нового ключа', () => {
        const data1 = getFixturePath('before_add.json');
        const data2 = getFixturePath('after_add.json');
        const diff = generateDiff(data1, data2);
        expect(diff).toMatchSnapshot();
    });

    test('удаление существовавшего ключа', () => {
        const data1 = getFixturePath('before_remove.json');
        const data2 = getFixturePath('after_remove.json');
        const diff = generateDiff(data1, data2);
        expect(diff).toMatchSnapshot();
    });

    test('изменение значения существующего ключа', () => {
        const data1 = getFixturePath('before_change.json');
        const data2 = getFixturePath('after_change.json');
        const diff = generateDiff(data1, data2);
        expect(diff).toMatchSnapshot();
    });

    test('изменение значения ключей и их порядка', () => {
        const data1 = getFixturePath('before_change_order.json');
        const data2 = getFixturePath('after_change_order.json');
        const diff = generateDiff(data1, data2);
        expect(diff).toMatchSnapshot();
    });

    test('без изменений', () => {
        const data1 = getFixturePath('before_no_change.json');
        const data2 = getFixturePath('after_no_change.json');
        const diff = generateDiff(data1, data2);
        expect(diff).toMatchSnapshot();
    });
});