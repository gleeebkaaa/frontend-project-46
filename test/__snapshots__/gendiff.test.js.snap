// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateDiff без изменений 1`] = `
"{
}"
`;

exports[`generateDiff добавление нового ключа 1`] = `
"{
+ c: 3
}"
`;

exports[`generateDiff изменение значения ключей и их порядка 1`] = `
"{
- a: 1
+ a: 2
- c: 4
+ d: 3
}"
`;

exports[`generateDiff изменение значения существующего ключа 1`] = `
"{
- a: 1
+ a: 2
}"
`;

exports[`generateDiff удаление существовавшего ключа 1`] = `
"{
- c: 3
}"
`;
